h2o-r/h2o-package/R/frame.R

```diff
@@ -2799,16 +2800,22 @@ destination_frame.guess <- function(x) {
 #'   stopifnot(is.h2o(hs), dim(hs)==dim(m))
 #' }
 #' }
-as.h2o <- function(x, destination_frame="", ...) {
-  .key.validate(destination_frame)
+as.h2o <- function(x, destination_frame=NULL, ...) {
+  if( !is.null(destination_frame) ) {
+    if( !(length(destination_frame)==1L && is.character(destination_frame)) )
+      stop("Argument 'destination_frame' must be a scalar character.")
+    .key.validate(destination_frame)
+    if( destination_frame %in% as.character(h2o.ls()$key) )
+      stop(paste0("Provided 'destination_frame' the H2O frame key '", destination_frame , "' already exists in your H2O Cluster. Please choose another key."))
```

Comment:
Why change the API with this check?
If a destination frame was specified, honor it.